core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -116,4 +123,67 @@ public static void checkRoleNamespace(Namespace namespace) {
         "Role namespace must have 3 levels, the input namespace is %s",
         namespace);
   }
+
+  // Install some post hooks used for ownership. The ownership will have the all privileges
+  // of securable objects, users, groups, roles.
+  public static <T> void prepareAuthorizationHooks(T manager, DispatcherHooks hooks) {
+    if (manager instanceof SupportsMetalakes) {
+      hooks.addPostHook(
+          "createMetalake",
```

Comment:
hi @jerqi  I think using function name string comparing is too error-prone.

We can create a `SetOwner` annotation to supports, for example:
```
public class MetalakeManager implements MetalakeDispatcher {
  @Override
  @SetOwner
  public BaseMetalake createMetalake(
      NameIdentifier ident, String comment, Map<String, String> properties)
      throws MetalakeAlreadyExistsException {
    long uid = idGenerator.nextId();
   ......
```

Use `SetOwner` annotation to comparing
```
if (method.isAnnotationPresent(SetOwner.class)) { 
NameIdentifier identifier = (NameIdentifier) ((Object[]) args)[0];
                            GravitinoEnv.getInstance()
                                    .ownershipManager()
                                    .setOwner(
                                            identifier,
                                            Entity.EntityType.METALAKE,
                                            PrincipalUtils.getCurrentUserName(),
                                            Owner.Type.USER);
}
```